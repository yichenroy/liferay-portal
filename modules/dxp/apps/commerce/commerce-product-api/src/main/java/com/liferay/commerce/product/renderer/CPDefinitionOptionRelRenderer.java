/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 * <p>
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 * <p>
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.renderer;

import aQute.bnd.annotation.ProviderType;

import com.liferay.portal.kernel.exception.PortalException;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

/**
 * @author Marco Leo
 */
@ProviderType
public interface CPDefinitionOptionRelRenderer {

	public String render(
		long cpDefinitionId, HttpServletRequest httpServletRequest,
		HttpServletResponse httpServletResponse) throws PortalException;

}