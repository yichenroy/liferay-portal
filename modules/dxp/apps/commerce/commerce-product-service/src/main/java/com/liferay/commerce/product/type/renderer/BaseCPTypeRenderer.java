/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.type.renderer;

import com.liferay.commerce.product.type.CPTypeRenderer;
import com.liferay.portal.kernel.log.Log;
import com.liferay.portal.kernel.log.LogFactoryUtil;
import com.liferay.portal.kernel.model.PortletConstants;
import com.liferay.portal.kernel.portlet.PortletBag;
import com.liferay.portal.kernel.portlet.PortletBagPool;
import com.liferay.portal.kernel.util.Portal;
import com.liferay.portal.kernel.util.Validator;
import com.liferay.portal.kernel.util.WebKeys;

import java.io.IOException;

import javax.servlet.RequestDispatcher;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
public abstract class BaseCPTypeRenderer implements CPTypeRenderer {

	public void renderJSP(
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		renderJSP(
			getServletContext(httpServletRequest), httpServletRequest,
			httpServletResponse, path);
	}

	public void renderJSP(
			ServletContext servletContext,
			HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse, String path)
		throws IOException {

		RequestDispatcher requestDispatcher =
			servletContext.getRequestDispatcher(path);

		try {
			requestDispatcher.include(httpServletRequest, httpServletResponse);
		}
		catch (ServletException se) {
			_log.error("Unable to render JSP " + path, se);

			throw new IOException("Unable to render " + path, se);
		}
	}

	public void setServletContext(ServletContext servletContext) {
		_servletContext = servletContext;
	}

	protected ServletContext getServletContext(
		HttpServletRequest httpServletRequest) {

		if (_servletContext != null) {
			return _servletContext;
		}

		String portletId = portal.getPortletId(httpServletRequest);

		if (Validator.isNotNull(portletId)) {
			String rootPortletId = PortletConstants.getRootPortletId(portletId);

			PortletBag portletBag = PortletBagPool.get(rootPortletId);

			return portletBag.getServletContext();
		}

		return (ServletContext)httpServletRequest.getAttribute(WebKeys.CTX);
	}

	@Reference
	protected Portal portal;

	private static final Log _log = LogFactoryUtil.getLog(
		BaseCPTypeRenderer.class);

	private ServletContext _servletContext;

}