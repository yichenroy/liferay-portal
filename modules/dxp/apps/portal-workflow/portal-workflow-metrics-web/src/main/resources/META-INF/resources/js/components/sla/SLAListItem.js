import autobind from 'autobind-decorator';
import { ChildLink } from '../../shared/components/router/routerWrapper';
import { formatDuration } from '../../shared/util/duration';
import Icon from '../../shared/components/Icon';
import React from 'react';
import SLAListCardContext from './SLAListCardContext';

class SLAListItem extends React.Component {
	@autobind
	showConfirmDialog() {
		const { id } = this.props;

		this.context.showConfirmDialog(id);
	}

	render() {
		const { description, duration, id, name, processId } = this.props;
		const durationString = formatDuration(duration);

		return (
			<tr>
				<td className="table-cell-expand">
					<div className="table-list-title">
						<ChildLink to={`/sla/edit/${processId}/${id}`}>{name}</ChildLink>
					</div>
				</td>

				<td>{description}</td>

				<td>{durationString}</td>

				<td>
					<div className="dropdown dropdown-action">
						<a
							aria-expanded="false"
							aria-haspopup="true"
							className="component-action dropdown-toggle"
							data-toggle="dropdown"
							href="#1"
							id="dropdownAction1"
							role="button"
						>
							<Icon iconName="ellipsis-v" />
						</a>

						<ul
							aria-labelledby=""
							className="dropdown-menu dropdown-menu-right"
						>
							<li>
								<ChildLink
									className="dropdown-item"
									to={`/sla/edit/${processId}/${id}`}
								>
									{Liferay.Language.get('edit')}
								</ChildLink>
							</li>

							<li>
								<button
									className="dropdown-item"
									onClick={this.showConfirmDialog}
								>
									{Liferay.Language.get('delete')}
								</button>
							</li>
						</ul>
					</div>
				</td>
			</tr>
		);
	}
}

SLAListItem.contextType = SLAListCardContext;
export default SLAListItem;