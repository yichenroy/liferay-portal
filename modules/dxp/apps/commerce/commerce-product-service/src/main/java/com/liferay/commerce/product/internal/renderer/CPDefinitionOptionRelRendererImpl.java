/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

package com.liferay.commerce.product.internal.renderer;

import com.liferay.commerce.product.model.CPDefinitionOptionRel;
import com.liferay.commerce.product.model.CPDefinitionOptionValueRel;
import com.liferay.commerce.product.renderer.CPDefinitionOptionRelRenderer;
import com.liferay.commerce.product.service.CPDefinitionOptionRelService;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderer;
import com.liferay.dynamic.data.mapping.form.renderer.DDMFormRenderingContext;
import com.liferay.dynamic.data.mapping.model.DDMForm;
import com.liferay.dynamic.data.mapping.model.DDMFormField;
import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
import com.liferay.dynamic.data.mapping.model.LocalizedValue;
import com.liferay.portal.kernel.exception.PortalException;
import com.liferay.portal.kernel.util.Portal;

import java.util.List;
import java.util.Locale;

import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;

import org.osgi.service.component.annotations.Component;
import org.osgi.service.component.annotations.Reference;

/**
 * @author Marco Leo
 */
@Component(immediate = true)
public class CPDefinitionOptionRelRendererImpl
	implements CPDefinitionOptionRelRenderer {

	@Override
	public String render(
			long cpDefinitionId, HttpServletRequest httpServletRequest,
			HttpServletResponse httpServletResponse)
		throws PortalException {

		DDMForm ddmForm = new DDMForm();

		Locale locale = _portal.getLocale(httpServletRequest);

		List<CPDefinitionOptionRel> cpDefinitionOptionRels =
			_cpDefinitionOptionRelService.
				getSkuContributorCPDefinitionOptionRels(cpDefinitionId);

		for (CPDefinitionOptionRel cpDefinitionOptionRel :
				cpDefinitionOptionRels) {

			List<CPDefinitionOptionValueRel> cpDefinitionOptionValueRels =
				cpDefinitionOptionRel.getCPDefinitionOptionValueRels();

			DDMFormField ddmFormField = new DDMFormField(
				String.valueOf(cpDefinitionOptionRel.
					getCPDefinitionOptionRelId()),
				cpDefinitionOptionRel.getDDMFormFieldTypeName());

			if (!cpDefinitionOptionValueRels.isEmpty()) {
				DDMFormFieldOptions ddmFormFieldOptions =
					new DDMFormFieldOptions();

				for (CPDefinitionOptionValueRel cpDefinitionOptionValueRel :
						cpDefinitionOptionValueRels) {

					ddmFormFieldOptions.addOptionLabel(
						String.valueOf(cpDefinitionOptionValueRel.
							getCPDefinitionOptionValueRelId()),
						locale, cpDefinitionOptionValueRel.getTitle(locale));
				}

				ddmFormField.setDDMFormFieldOptions(ddmFormFieldOptions);
			}

			LocalizedValue localizedValue = new LocalizedValue(locale);

			localizedValue.addString(
				locale, cpDefinitionOptionRel.getName(locale));

			ddmFormField.setLabel(localizedValue);

			ddmFormField.setRequired(true);

			ddmForm.addDDMFormField(ddmFormField);
		}

		DDMFormRenderingContext ddmFormRenderingContext =
			new DDMFormRenderingContext();

		ddmFormRenderingContext.setContainerId("options");
		ddmFormRenderingContext.setHttpServletRequest(httpServletRequest);
		ddmFormRenderingContext.setHttpServletResponse(httpServletResponse);
		ddmFormRenderingContext.setLocale(locale);

		return _ddmFormRenderer.render(ddmForm, ddmFormRenderingContext);
	}

	@Reference
	private CPDefinitionOptionRelService _cpDefinitionOptionRelService;

	@Reference
	private DDMFormRenderer _ddmFormRenderer;

	@Reference
	private Portal _portal;

}