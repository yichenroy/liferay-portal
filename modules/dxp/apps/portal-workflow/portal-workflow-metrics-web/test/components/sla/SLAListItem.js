import React from 'react';
import renderer from 'react-test-renderer';
import {MockRouter as Router} from 'test/mock/MockRouter';
import SLAListItem from 'components/sla/SLAListItem';

test('Should render component', () => {
	const component = renderer.create(
		<Router>
			<SLAListItem
				id={1234}
				instancesCount='10'
				onTime='5'
				overdue='5'
				processName='Process test'
			/>
		</Router>
	);

	const tree = component.toJSON();

	expect(tree).toMatchSnapshot();
});

test('Should render component', () => {
	const component = mount(
		<Router>
			<SLAListItem
				instancesCount='10'
				onTime='5'
				overdue='5'
				processName='Process test'
			/>
		</Router>
	);

	const instance = component.find(SLAListItem).instance();

	instance.context = {
		showConfirmDialog: () => {}
	};

	instance.showConfirmDialog();

	expect(component).toMatchSnapshot();
});

test('Should render component blocked', () => {
	const component = mount(
		<Router>
			<SLAListItem
				dateModified={
					new Date(Date.UTC('2019', '04', '06', '20', '32', '18'))
				}
				instancesCount='10'
				onTime='5'
				overdue='5'
				processName='Process test'
				status={2}
			/>
		</Router>
	);

	const instance = component.find(SLAListItem).instance();

	instance.context = {
		showConfirmDialog: () => {}
	};

	instance.showConfirmDialog();

	expect(component).toMatchSnapshot();
});
