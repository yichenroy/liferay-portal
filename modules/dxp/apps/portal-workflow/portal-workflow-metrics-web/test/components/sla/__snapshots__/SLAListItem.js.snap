// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component 1`] = `
<tr>
  <td
    className="table-cell-expand"
  >
    <div
      className="table-list-title"
    >
       
      <a
        href="/sla/edit/undefined/1234?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
        onClick={[Function]}
      />
    </div>
  </td>
  <td />
  <td
    className=""
  >
    running
  </td>
  <td>
    
  </td>
  <td>
    Aug 27
  </td>
  <td>
    <div
      className="dropdown dropdown-action"
    >
      <a
        aria-expanded="false"
        aria-haspopup="true"
        className="component-action dropdown-toggle"
        data-toggle="dropdown"
        href="#1"
        id="dropdownAction1"
        role="button"
      >
        <svg
          className="lexicon-icon lexicon-icon-ellipsis-v"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<use xlink:href=\\"http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#ellipsis-v\\" />",
            }
          }
          focusable="false"
          role="presentation"
        />
      </a>
      <ul
        aria-labelledby=""
        className="dropdown-menu dropdown-menu-right"
      >
        <li>
          <a
            className="dropdown-item"
            href="/sla/edit/undefined/1234?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
            onClick={[Function]}
          >
            edit
          </a>
        </li>
        <li>
          <button
            className="dropdown-item"
            onClick={[Function]}
          >
            delete
          </button>
        </li>
      </ul>
    </div>
  </td>
</tr>
`;

exports[`Should render component 2`] = `
<MockRouter>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SLAListItem
        instancesCount="10"
        onTime="5"
        overdue="5"
        processName="Process test"
      >
        <tr>
          <td
            className="table-cell-expand"
          >
            <div
              className="table-list-title"
            >
               
              <withRouter()
                to="/sla/edit/undefined/undefined"
              >
                <Component
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  to="/sla/edit/undefined/undefined"
                >
                  <ChildLinkWrapper
                    backPath="/"
                    currentPath="/processes/1/10/title%3Aasc?backPath=%2F"
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "canGo": [Function],
                        "createHref": [Function],
                        "entries": Array [
                          Object {
                            "hash": "",
                            "match": Object {
                              "params": Object {
                                "page": 1,
                                "search": undefined,
                                "sort": undefined,
                              },
                            },
                            "pathname": "/processes/1/10/title%3Aasc",
                            "search": "?backPath=%2F",
                          },
                        ],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "index": 0,
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    to="/sla/edit/undefined/undefined"
                  >
                    <Link
                      to={
                        Object {
                          "pathname": "/sla/edit/undefined/undefined",
                          "search": "?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F",
                        }
                      }
                    >
                      <a
                        href="/sla/edit/undefined/undefined?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                        onClick={[Function]}
                      />
                    </Link>
                  </ChildLinkWrapper>
                </Component>
              </withRouter()>
            </div>
          </td>
          <td />
          <td
            className=""
          >
            running
          </td>
          <td />
          <td>
            Aug 27
          </td>
          <td>
            <div
              className="dropdown dropdown-action"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                className="component-action dropdown-toggle"
                data-toggle="dropdown"
                href="#1"
                id="dropdownAction1"
                role="button"
              >
                <Icon
                  iconName="ellipsis-v"
                >
                  <svg
                    className="lexicon-icon lexicon-icon-ellipsis-v"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<use xlink:href=\\"http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#ellipsis-v\\" />",
                      }
                    }
                    focusable="false"
                    role="presentation"
                  />
                </Icon>
              </a>
              <ul
                aria-labelledby=""
                className="dropdown-menu dropdown-menu-right"
              >
                <li>
                  <withRouter()
                    className="dropdown-item"
                    to="/sla/edit/undefined/undefined"
                  >
                    <Component
                      className="dropdown-item"
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "canGo": [Function],
                          "createHref": [Function],
                          "entries": Array [
                            Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                          ],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "index": 0,
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "match": Object {
                              "params": Object {
                                "page": 1,
                                "search": undefined,
                                "sort": undefined,
                              },
                            },
                            "pathname": "/processes/1/10/title%3Aasc",
                            "search": "?backPath=%2F",
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        }
                      }
                      match={
                        Object {
                          "isExact": false,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      to="/sla/edit/undefined/undefined"
                    >
                      <ChildLinkWrapper
                        backPath="/"
                        className="dropdown-item"
                        currentPath="/processes/1/10/title%3Aasc?backPath=%2F"
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "canGo": [Function],
                            "createHref": [Function],
                            "entries": Array [
                              Object {
                                "hash": "",
                                "match": Object {
                                  "params": Object {
                                    "page": 1,
                                    "search": undefined,
                                    "sort": undefined,
                                  },
                                },
                                "pathname": "/processes/1/10/title%3Aasc",
                                "search": "?backPath=%2F",
                              },
                            ],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "index": 0,
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        to="/sla/edit/undefined/undefined"
                      >
                        <Link
                          className="dropdown-item"
                          to={
                            Object {
                              "pathname": "/sla/edit/undefined/undefined",
                              "search": "?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F",
                            }
                          }
                        >
                          <a
                            className="dropdown-item"
                            href="/sla/edit/undefined/undefined?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                            onClick={[Function]}
                          >
                            edit
                          </a>
                        </Link>
                      </ChildLinkWrapper>
                    </Component>
                  </withRouter()>
                </li>
                <li>
                  <button
                    className="dropdown-item"
                    onClick={[Function]}
                  >
                    delete
                  </button>
                </li>
              </ul>
            </div>
          </td>
        </tr>
      </SLAListItem>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should render component blocked 1`] = `
<MockRouter>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <SLAListItem
        dateModified={2019-05-06T20:32:18.000Z}
        instancesCount="10"
        onTime="5"
        overdue="5"
        processName="Process test"
        status={2}
      >
        <tr>
          <td
            className="table-cell-expand"
          >
            <div
              className="table-list-title"
            >
              <Icon
                elementClasses="text-danger"
                iconName="exclamation-full"
              >
                <svg
                  className="lexicon-icon lexicon-icon-exclamation-full text-danger"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "<use xlink:href=\\"http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#exclamation-full\\" />",
                    }
                  }
                  focusable="false"
                  role="presentation"
                />
              </Icon>
               
              <withRouter()
                to="/sla/edit/undefined/undefined"
              >
                <Component
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "canGo": [Function],
                      "createHref": [Function],
                      "entries": Array [
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                      ],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "index": 0,
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "match": Object {
                          "params": Object {
                            "page": 1,
                            "search": undefined,
                            "sort": undefined,
                          },
                        },
                        "pathname": "/processes/1/10/title%3Aasc",
                        "search": "?backPath=%2F",
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  location={
                    Object {
                      "hash": "",
                      "match": Object {
                        "params": Object {
                          "page": 1,
                          "search": undefined,
                          "sort": undefined,
                        },
                      },
                      "pathname": "/processes/1/10/title%3Aasc",
                      "search": "?backPath=%2F",
                    }
                  }
                  match={
                    Object {
                      "isExact": false,
                      "params": Object {},
                      "path": "/",
                      "url": "/",
                    }
                  }
                  to="/sla/edit/undefined/undefined"
                >
                  <ChildLinkWrapper
                    backPath="/"
                    currentPath="/processes/1/10/title%3Aasc?backPath=%2F"
                    history={
                      Object {
                        "action": "POP",
                        "block": [Function],
                        "canGo": [Function],
                        "createHref": [Function],
                        "entries": Array [
                          Object {
                            "hash": "",
                            "match": Object {
                              "params": Object {
                                "page": 1,
                                "search": undefined,
                                "sort": undefined,
                              },
                            },
                            "pathname": "/processes/1/10/title%3Aasc",
                            "search": "?backPath=%2F",
                          },
                        ],
                        "go": [Function],
                        "goBack": [Function],
                        "goForward": [Function],
                        "index": 0,
                        "length": 1,
                        "listen": [Function],
                        "location": Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        },
                        "push": [Function],
                        "replace": [Function],
                      }
                    }
                    to="/sla/edit/undefined/undefined"
                  >
                    <Link
                      to={
                        Object {
                          "pathname": "/sla/edit/undefined/undefined",
                          "search": "?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F",
                        }
                      }
                    >
                      <a
                        href="/sla/edit/undefined/undefined?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                        onClick={[Function]}
                      />
                    </Link>
                  </ChildLinkWrapper>
                </Component>
              </withRouter()>
            </div>
          </td>
          <td />
          <td
            className="text-danger"
          >
            blocked
          </td>
          <td />
          <td>
            May 06
          </td>
          <td>
            <div
              className="dropdown dropdown-action"
            >
              <a
                aria-expanded="false"
                aria-haspopup="true"
                className="component-action dropdown-toggle"
                data-toggle="dropdown"
                href="#1"
                id="dropdownAction1"
                role="button"
              >
                <Icon
                  iconName="ellipsis-v"
                >
                  <svg
                    className="lexicon-icon lexicon-icon-ellipsis-v"
                    dangerouslySetInnerHTML={
                      Object {
                        "__html": "<use xlink:href=\\"http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#ellipsis-v\\" />",
                      }
                    }
                    focusable="false"
                    role="presentation"
                  />
                </Icon>
              </a>
              <ul
                aria-labelledby=""
                className="dropdown-menu dropdown-menu-right"
              >
                <li>
                  <withRouter()
                    className="dropdown-item"
                    to="/sla/edit/undefined/undefined"
                  >
                    <Component
                      className="dropdown-item"
                      history={
                        Object {
                          "action": "POP",
                          "block": [Function],
                          "canGo": [Function],
                          "createHref": [Function],
                          "entries": Array [
                            Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                          ],
                          "go": [Function],
                          "goBack": [Function],
                          "goForward": [Function],
                          "index": 0,
                          "length": 1,
                          "listen": [Function],
                          "location": Object {
                            "hash": "",
                            "match": Object {
                              "params": Object {
                                "page": 1,
                                "search": undefined,
                                "sort": undefined,
                              },
                            },
                            "pathname": "/processes/1/10/title%3Aasc",
                            "search": "?backPath=%2F",
                          },
                          "push": [Function],
                          "replace": [Function],
                        }
                      }
                      location={
                        Object {
                          "hash": "",
                          "match": Object {
                            "params": Object {
                              "page": 1,
                              "search": undefined,
                              "sort": undefined,
                            },
                          },
                          "pathname": "/processes/1/10/title%3Aasc",
                          "search": "?backPath=%2F",
                        }
                      }
                      match={
                        Object {
                          "isExact": false,
                          "params": Object {},
                          "path": "/",
                          "url": "/",
                        }
                      }
                      to="/sla/edit/undefined/undefined"
                    >
                      <ChildLinkWrapper
                        backPath="/"
                        className="dropdown-item"
                        currentPath="/processes/1/10/title%3Aasc?backPath=%2F"
                        history={
                          Object {
                            "action": "POP",
                            "block": [Function],
                            "canGo": [Function],
                            "createHref": [Function],
                            "entries": Array [
                              Object {
                                "hash": "",
                                "match": Object {
                                  "params": Object {
                                    "page": 1,
                                    "search": undefined,
                                    "sort": undefined,
                                  },
                                },
                                "pathname": "/processes/1/10/title%3Aasc",
                                "search": "?backPath=%2F",
                              },
                            ],
                            "go": [Function],
                            "goBack": [Function],
                            "goForward": [Function],
                            "index": 0,
                            "length": 1,
                            "listen": [Function],
                            "location": Object {
                              "hash": "",
                              "match": Object {
                                "params": Object {
                                  "page": 1,
                                  "search": undefined,
                                  "sort": undefined,
                                },
                              },
                              "pathname": "/processes/1/10/title%3Aasc",
                              "search": "?backPath=%2F",
                            },
                            "push": [Function],
                            "replace": [Function],
                          }
                        }
                        to="/sla/edit/undefined/undefined"
                      >
                        <Link
                          className="dropdown-item"
                          to={
                            Object {
                              "pathname": "/sla/edit/undefined/undefined",
                              "search": "?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F",
                            }
                          }
                        >
                          <a
                            className="dropdown-item"
                            href="/sla/edit/undefined/undefined?backPath=%2Fprocesses%2F1%2F10%2Ftitle%253Aasc%3FbackPath%3D%252F"
                            onClick={[Function]}
                          >
                            edit
                          </a>
                        </Link>
                      </ChildLinkWrapper>
                    </Component>
                  </withRouter()>
                </li>
                <li>
                  <button
                    className="dropdown-item"
                    onClick={[Function]}
                  >
                    delete
                  </button>
                </li>
              </ul>
            </div>
          </td>
        </tr>
      </SLAListItem>
    </Router>
  </MemoryRouter>
</MockRouter>
`;
